### ISSUE TITLE
utcToZonedTime() incorrect while toDate() correct around daylight saving time

### ISSUE THREAD

**Issue by soetji** [marnusw/date-fns-tz#31]
This year for "America/New_York" the daylight saving ended on Nov 3, 02:00 where it turned back 01:00. My browser is on a local "America/New_York" system.

toDate() gets this correctly but not utcToZonedTime(). utcToZonedTime() adds an hour even before 02:00.

```
toDate('2019-11-03T03:00:00.000Z', { timeZone: 'America/New_York' })
// Sat Nov 02 2019 23:00:00 GMT-0400 (Eastern Daylight Time)

utcToZonedTime('2019-11-03T03:00:00.000Z', 'America/New_York')
// Sun Nov 03 2019 00:00:00 GMT-0400 (Eastern Daylight Time)


toDate('2019-11-03T04:00:00.000Z', { timeZone: 'America/New_York' })
// Sun Nov 03 2019 00:00:00 GMT-0400 (Eastern Daylight Time)

utcToZonedTime('2019-11-03T04:00:00.000Z', 'America/New_York')
// Sun Nov 03 2019 01:00:00 GMT-0400 (Eastern Daylight Time)


toDate('2019-11-03T05:00:00.000Z', { timeZone: 'America/New_York' })
// Sun Nov 03 2019 01:00:00 GMT-0400 (Eastern Daylight Time)

utcToZonedTime('2019-11-03T05:00:00.000Z', 'America/New_York')
// Sun Nov 03 2019 01:00:00 GMT-0500 (Eastern Standard Time)


toDate('2019-11-03T06:00:00.000Z', { timeZone: 'America/New_York' })
// Sun Nov 03 2019 01:00:00 GMT-0500 (Eastern Standard Time)

utcToZonedTime('2019-11-03T06:00:00.000Z', 'America/New_York')
// Sun Nov 03 2019 01:00:00 GMT-0500 (Eastern Standard Time)


toDate('2019-11-03T07:00:00.000Z', { timeZone: 'America/New_York' })
// Sun Nov 03 2019 02:00:00 GMT-0500 (Eastern Standard Time)

utcToZonedTime('2019-11-03T07:00:00.000Z', 'America/New_York')
// Sun Nov 03 2019 02:00:00 GMT-0500 (Eastern Standard Time)
```

**Comment by soetji** [comment]
It looks like the bug in utcToZonedTime goes with the system timezone. It works correctly on the GMT timezone system.

```
// On a GMT timezone system
utcToZonedTime(1572750000000, 'America/New_York')
// Sat Nov 02 2019 23:00:00 GMT-0400 (Eastern Daylight Time)
utcToZonedTime(1572750000000, 'America/New_York').getTimezoneOffset()
// 0

// On a America/New_York timezone system
utcToZonedTime(1572750000000, 'America/New_York')
// Sun Nov 03 2019 00:00:00 GMT-0400 (Eastern Daylight Time)
utcToZonedTime(1572750000000, 'America/New_York').getTimezoneOffset()
// 300

**Comment by marnusw** [comment]
That's good to know! I'm on GMT so would probably not have managed to reproduce this.

**Comment by constgen** [comment]
This is a real problem. `utcToZonedTime()` gives a different result when you use the system timeZone and pass the date near the DST switching hour

```js
let timeZone = new Intl.DateTimeFormat().resolvedOptions().timeZone
let date = new Date() // choose the date +1h right after switching to Standart Time from Summer Time
let zonedDate = utcToZonedTime(value, timeZone);
let time = format(zonedDate, 'p', { timeZone })
console.log(time) // As the time zone is system, the time should match to `date` time, but it is not.
```

Actually this formatting produces the same hour 3 times during moving from Summer Time to Standart Time

**Comment by altodoro** [comment]
Seems like the similar issue here: **utcToZonedTime()** adds an extra day to the returned date when converting date with time in range 23:00 - 23:59
```
utcToZonedTime('2020-02-26T23:15:00Z', 'Europe/Berlin');
// expected -> Fri Feb 27 2020 00:15:00 GMT+0200 (Eastern European Standard Time)
// actual   -> Fri Feb 28 2020 00:15:00 GMT+0200 (Eastern European Standard Time)
```

**Comment by marnusw** [comment]
@altodoro I think the issue you are reporting is #38 / #43 rather than related to issues around the DST change. Please check whether version `1.0.10` solves your issue.

**Comment by altodoro** [comment]
> @altodoro I think the issue you are reporting is #38 / #43 rather than related to issues around the DST change. Please check whether version `1.0.10` solves your issue.

yep, you're right. Thanks for the update!

**Comment by marnusw** [comment]
This might be fixed in version `1.0.11` which will be available soon. Please reopen if not.
